src/common/app_ble_gap.cpp

```diff
@@ -455,4 +456,50 @@ uint32_t app_ble_gap_adv_set_unregister(uint8_t adv_handle, uint8_t **pp_adv_dat
         return NRF_ERROR_SD_RPC_INVALID_STATE;
     }
 }
+
+int app_ble_gap_adv_buf_register(void * p_buf)
+{
+    if (!app_ble_gap_check_current_adapter_set(REQUEST_REPLY_CODEC_CONTEXT))
+    {
+        return NRF_ERROR_SD_RPC_INVALID_STATE;
+    }
+
+    try
+    {
+        const auto gap_state = adapters_gap_state.at(current_request_reply_context.adapter_id);
+
+        int id = 1;
+        for (auto &addr : gap_state->ble_gap_adv_buf_addr)
+        {
+            if (addr == nullptr)
+            {
+                addr = p_buf;
+                return id;
+            }
+            id++;
+        }
+
+        return 0;
+    }
+
+    catch (const std::out_of_range &)
+    {
+        return 0;
+    }
+}
+
+void *app_ble_gap_adv_buf_unregister(const int id)
+{
+    if (!app_ble_gap_check_current_adapter_set(EVENT_CODEC_CONTEXT))
```

Comment:
Can we ensure that this function is only called in EVENT_CODEC_CONTEXT?

Reply:
that may not be true. eg it should be unregistered if `scan_start` returns error (it may not be done now)

Reply:
started to think about `scan_stop` case. Will there be any event that will unregister the buffer? If not then this case is not handled.